### PR TITLE
Take compression into account for cache upload/download byte metrics

### PR DESCRIPTION
**Pull request by maggie-lou**
Before, we always read/wrote decompressed bytes from the cache, so we could use the digest size as the number of bytes uploaded/downloaded from the cache. Now that the cache can read/write decompressed bytes, we need to modify the metric to use the actual number of bytes transferred.

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3182** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/3182" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/3182?utm_source=stack-comment).

**Comment by siggisim** [comment]
Does this affect the Usages table?

**Comment by siggisim** [comment]
If so - we should probably track uncompressed vs compressed bytes separately, as it'll affect billing.

**Comment by maggie-lou** [comment]
> Does this affect the Usages table?

No - it will only change the grafana metrics remote_cache.download_size_bytes and remote_cache.upload_size_bytes

File: server/util/compression/compression.go [hunk 1]
```diff
@@ -138,6 +148,9 @@ func (d *zstdDecompressor) Close() error {
 // at least a few hundred bytes.
 func NewZstdCompressingReader(reader io.Reader, readBuf []byte, compressBuf []byte) (io.ReadCloser, error) {
```
Comment (bduffany) [inline server/util/compression/compression.go:149]: would something like this work ?

```
counter := ioutil.Counter{}
compressed := compression.NewZstdCompressingReader(io.TeeReader(counter, reader), ...)
doSomethingWith(counter.Count())
```

**Review by tylerwilliams — APPROVED** [review]
Thanks for considering the alternative

File: server/util/compression/compression.go (hunk 1 above)
Comment (maggie-lou) [reply server/util/compression/compression.go]: Nice! Done. @bduffany could you look over the final version when you have a chance?

**Pull request merged**